runtime/autoload/help.vim

```diff
@@ -0,0 +1,38 @@
+function! s:synnames(...) abort
+  if a:0
+    let [line, col] = [a:1, a:2]
+  else
+    let [line, col] = [line('.'), col('.')]
+  endif
+  return reverse(map(synstack(line, col), 'synIDattr(v:val,"name")'))
+endfunction
+
+function! HelpTopic()
+  let col = col('.') - 1
+  while col && getline('.')[col] =~# '\k'
+    let col -= 1
+  endwhile
+  let pre = col == 0 ? '' : getline('.')[0 : col]
+  let col = col('.') - 1
+  while col && getline('.')[col] =~# '\k'
+    let col += 1
+  endwhile
+  let post = getline('.')[col : -1]
+  let syn = get(synnames(), 0, '')
```

Comment:
Missing `s:` before `synnames`. Also you do *not* need this function *at all*: simply use `synIDattr(synID(line('.'), col('.')), 'name')`, you are using only very small part of the `s:synnames` functionality.